core/domain/exp_jobs_one_off.py

```diff
@@ -728,3 +728,179 @@ def map(item):
     @staticmethod
     def reduce(key, values):
         yield (key, values)
+
+
+class ExpSnapshotsMigrationAuditJob(jobs.BaseMapReduceOneOffJobManager):
+    """A reusable one-off job for testing the migration of all exp versions
+    to the latest schema version. This job runs the state migration, but does
+    not commit the new exploration to the datastore.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationSnapshotContentModel]
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        super(ExpSnapshotsMigrationAuditJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @staticmethod
+    def map(item):
+        exp_id = item.get_unversioned_instance_id()
+
+        latest_exploration = exp_fetchers.get_exploration_by_id(
+            exp_id, strict=False)
+        if latest_exploration is None:
+            yield ('INFO - Exploration does not exist', item.id)
+            return
+
+        exploration_model = exp_models.ExplorationModel.get(exp_id)
+        if (exploration_model.states_schema_version !=
+                feconf.CURRENT_STATE_SCHEMA_VERSION):
+            yield (
+                'FAILURE - Exploration is not at latest schema version', exp_id)
+            return
+
+        try:
+            latest_exploration.validate()
+        except Exception as e:
+            yield (
+                'INFO - Exploration %s failed non-strict validation' % item.id,
+                e)
+
+        # Some old explorations do not have a states schema version.
+        if 'states_schema_version' not in item.content:
+            item.content['states_schema_version'] = 0
```

Comment:
nit: `None` seems like a more appropriate value here?

Reply:
No, this must be 0. All items should have this field, and this handles old explorations that were created before the states_schema_version system was introduced.

If I end up making a follow-up commit to this PR, I will add more documentation here to clarify.

Reply:
I think that's worth explicitly clarifying since it's a rather esoteric state that's unexpected in the current world of things where a schema version is well-understood.

Reply:
Done, added a comment. Thanks!

Reply:
This all SGTM, thanks!